nussl/core/audio_signal.py

```diff
@@ -1644,6 +1649,311 @@ def embed_audio(self, ext='.mp3', display=True):
         from . import play_utils
         return play_utils.embed_audio(self, ext=ext, display=display)
 
+    ##################################
+    #          Effect Hooks          #
+    ##################################
+
+    def build_effects(self, reset=True, overwrite=False):
```

Comment:
Does this apply effects to my audio or is this a preprocessing step before I can apply effects? If the former, this method should be renamed to `apply_effects`, if the latter then the documentation needs to be clearer.

Reply:
This applies effects. Changed function name